featurebyte/service/historical_features_and_target.py

```diff
@@ -298,6 +298,10 @@ async def get_target(
     request_table_name = f"{REQUEST_TABLE_NAME}_{request_id}"
     request_table_columns = observation_set.columns
 
+    # filter request columns if nodes request them as well
+    nodes_columns = [graph.get_node_output_column_name(node.name) for node in nodes]
+    request_table_columns = [col for col in request_table_columns if col not in nodes_columns]
```

Comment:
I think this issue can occur for any observation table which happen to have a column which have same name as a feature. 

For features it is less common and only possible if users will explicitly add some column which will repeat the name of the feature, and can probably be classified as user error. 

For target it is a bit more tricky. We have a functionality to "Compute New Observation table with target value", i.e. create observation table by adding target column to an existing observation table. 
In this case it cannot be classified as user error as user does not explicitly add any columns to observation table, this happens behind the scenes. 

Regarding caching, Feature Table Cache is designed around definition hash of nodes now, all the caching is based around graph and nodes. 
Columns in observation table are not available as nodes, we operate on them separately as a list of `request_table_columns` (at least to my knowledge). 
Thus, if I decide to skip a target `node`, I have **nothing to cache**, because for target we have only one node - target node - which we what to skip.